src/components/chat/tabs/contract-tab/termination-date.tsx

```
@@ -0,0 +1,222 @@
+                          header: 'Termination reverted',
+                          message: 'Successfully reverted the termination',
+                        })
+                        reset()
+                      })
+                      .catch((error) => {
+                        showNotification({
+                          type: 'red',
+                          header: 'Unable to revert termination',
+                          message: error.message,
+                        })
+                      })
+                  }
+                }}
+              >
+                Revert termination
+              </Button>
+            </ButtonsGroup>
+          </>
+        )}
+        {datePickerEnabled && (
+          <>
+            <DateTimePicker
+              date={terminationDate}
+              setDate={setTerminationDate}
+            />
+            <ButtonsGroup>
+              <Button
+                fullWidth
+                variation={'primary'}
+                disabled={changeTerminationDateLoading}
+                onClick={() => {
+                  const confirmed = window.confirm(
+                    `Are you sure you want to change the termination date from ${
+                      contract.terminationDate
+                    } to ${format(terminationDate, 'yyyy-MM-dd')}?`,
+                  )
+                  if (confirmed) {
+                    changeTerminationDate(
+                      changeTerminationDateOptions(contract, terminationDate),
+                    )
+                      .then(() => {
+                        showNotification({
+                          type: 'olive',
+                          header: 'Termination date changed',
+                          message: 'Successfully changed termination date.',
+                        })
+                        reset()
+                      })
+                      .catch((error) => {
+                        showNotification({
```

Just a thought in these catch blocks, shall we rethrow the error, or at least write it to the console, or capture it with sentry, or something? Just to not "swallow" it? (applies to all these catch blocks i guess)

---

Throwing them again after the notification